build-service.py

```diff
@@ -88,73 +101,60 @@ def load_config_file(filename):
     return config
 
 
-def build_repo(repo, branch, deploy_conf, service_conf):
-    # do work in temp dir
-    logging.debug(
-        f"building repo '{repo}@{branch}' with deploy conf '{deploy_conf}' and service conf '{service_conf}'"
-    )
+def get_deploy_conf(deploy_conf, repo_dir):
+    # parse options from default config if none given
+    if deploy_conf is None:
+        deploy_conf = "kaas.deploy.yml"
+    if isinstance(deploy_conf, str):
+        return load_config_file(f"{repo_dir}/{deploy_conf}")
 
-    with tempfile.TemporaryDirectory(prefix="kaas-repo-build-") as repo_dir:
 
-        logging.info(f"cloning repo {repo} to {repo_dir}")
+def get_service_conf(service_conf, repo_dir):
+    # parse options from default config if none given
+    if service_conf is None:
+        service_conf = "kaas.service.yml"
+    if isinstance(service_conf, str):  # if string, its a filepath to load from
+        return load_config_file(f"{repo_dir}/{service_conf}")
 
-        # add empty creds to clone url
-        # git asks for auth if repo isnt public
-        # we're only supporting public repos for now
-        repo = re.sub(r"^https?://(:@)?", "https://:@", repo)
 
-        try:
-            repo = git.Repo.clone_from(repo, repo_dir)
-        except git.exc.GitCommandError:
-            raise BadGitRepo
-
-        if branch is not None:
-            try:
-                repo.git.checkout(branch)
-            except git.exc.GitCommandError:
-                raise BadGitBranch(branch)
+def build_repo(repo_dir, deploy_conf, service_conf):
```

Comment:
Did you mean to remove the `repo_branch` arg / move the repo cloning to `__main__`? The repo cloning was handled in `build_repo()` as to not duplicate code between the CLI and API funcs.

Reply:
addressed this in new PR